tests/integration/targets/vmware_host_iscsi/tasks/main.yml

```
@@ -9,15 +9,17 @@
 
 - name: prepare of integration test for iSCSI - connect to vCenter
   block:
-    - vmware_host_iscsi:
+    - name: Enable iSCSI
+      community.vmware.vmware_host_iscsi:
         hostname: "{{ vcenter_hostname }}"
         username: "{{ vcenter_username }}"
         password: "{{ vcenter_password }}"
         validate_certs: false
         esxi_hostname: "{{ esxi1 }}"
         state: enabled
```

```suggestion
        state: enabled
      register: create_iscsi_device_result
```

Instead of discovering the iSCSI adapter with `vmware_host_facts` or `vmware_host_iscsi_info`, could'nt we use the result from this task? If we register this as `create_iscsi_device_result`, I think we should find the adapter in `create_iscsi_device_result.iscsi_properties.vmhba_name`.

---

Thanks, @mariolenz for the advice.
I reflected on your comment.